### PR TITLE
Add QinQ subinterface

### PR DESCRIPTION
**Pull request by sojindal**
Add subinterface config with QinQ encapsulation

### CODE REVIEW

**Comment by sumukhatv** [comment]
The PR is missing the API implementation and tests

File: sonic_api.yaml [hunk 1]
```diff
@@ -1400,3 +1679,17 @@ definitions:
...
+        type: string
+      vrf_id:
+        type: string
+        description: id representing the virtual routing instance (VRF) this Subinterface is associated to.
+      ip_addr:
+        type: string
+        description: subinterface ip address
+      mask:
+        type: string
+        description: subinterface mask
```
Comment (lguohan) [inline sonic_api.yaml:1695]: subinterface ip mask

Comment (sojindal) [reply sonic_api.yaml:1695]: Thanks for pointing! Will update 👍

File: sonic_api.yaml [hunk 2]
```diff
@@ -1255,6 +1255,285 @@ paths:
           schema:
             $ref: '#/definitions/Error'
 #----------------------------------------------
+# QinQ subinterface object
+#----------------------------------------------
+  '/config/subinterface/qinq/{if_name}/{outer_tag}/{inner_tag}':
+    put:
+      summary: Create/update QinQ router interface for a physical interface
+      description: If the interface is already created, only QinQ attributes can be updated.
```
Comment (prsunny) [inline sonic_api.yaml:1263]: Please mention what attributes can be modifed

Comment (sojindal) [reply sonic_api.yaml:1263]: I don't see anything which cannot be updated in QinQEntry object. Exceptions can be added in implementation for lets say if some field updation is dependent on some other resources.

**Review by prsunny — APPROVED** [review]

File: sonic_api.yaml [hunk 3]
```diff
@@ -1400,3 +1679,17 @@ definitions:
...
         description: IP address of directly attached device to the switch
+  QinQEntry:
+    type: object
+    properties:
+      description:
+        type: string
+      vrf_id:
+        type: string
+        description: id representing the virtual routing instance (VRF) this Subinterface is associated to.
+      ip_addr:
```
Comment (lguohan) [inline sonic_api.yaml:1690]: is this only for ipv4?

File: sonic_api.yaml [hunk 4]
```diff
@@ -1400,3 +1679,17 @@ definitions:
...
+        type: string
+        description: subinterface ip address
```
Comment (lguohan) [inline sonic_api.yaml:1692]: ipv4

File: sonic_api.yaml [hunk 5]
```diff
@@ -1400,3 +1679,17 @@ definitions:
...
+      mask:
+        type: string
+        description: subinterface ip mask
```
Comment (lguohan) [inline sonic_api.yaml:1695]: ipv4 only.

File: sonic_api.yaml (hunk 3 above)
Comment (sojindal) [reply sonic_api.yaml:1690]: Yes

File: sonic_api.yaml (hunk 4 above)
Comment (sojindal) [reply sonic_api.yaml]: Updated

File: sonic_api.yaml (hunk 5 above)
Comment (sojindal) [reply sonic_api.yaml]: Updated

**Pull request merged**